Keas.Mvc/Controllers/TeamAdminController.cs

```diff
@@ -746,19 +752,304 @@ public async Task<IActionResult> Upload(IFormFile file)
+            var rowNumber = 1;
+            bool import = true;
+            bool somethingSaved = false;
+
+            if (file == null || file.Length == 0)
+            {
+                Message = "File not selected";
+                return RedirectToAction("Upload");
+            }
+
+            // Add counts
+            var team = await _context.Teams.FirstAsync(t => t.Slug == Team);
+            using (var reader = new StreamReader(file.OpenReadStream()))
+            using (var csv = new CsvReader(reader))
+            {
+                csv.Configuration.PrepareHeaderForMatch = (string header, int index) => header.ToLower();
+                var record = new EquipmentImport();
+                var records = csv.EnumerateRecords(record);
+                foreach (var r in records)
+                {
+                    var recEquipmentCount = 0;
+                    var recAttributeCount = 0;
+                    var recAttributeAdded = false;
+                    var recPeopleCount = 0;
+                    var recAssignmentCount = 0;
+
+                    using (var transaction = await _context.Database.BeginTransactionAsync())
+                    {
+                        somethingSaved = false;
+                        rowNumber += 1;
+                        var result = new EquipmentImportResults(r)
+                        {
+                            LineNumber = rowNumber,
+                            Success = true
+                        };
+                        ModelState.Clear();
+
+                        if (!string.IsNullOrWhiteSpace(r.EquipmentName) && !r.Make.Contains("AEXMPLE"))
+                        {
+                            var equipment = new Equipment();
+                            equipment.Name = r.EquipmentName;
+                            equipment.SerialNumber = r.SerialNumber;
+                            equipment.Make = r.Make;
+                            equipment.Model = r.Model;
+                            equipment.Tags = string.IsNullOrWhiteSpace(r.Tag) ? "Imported" : $"{r.Tag}, Imported";
+                            equipment.TeamId = team.Id;
+                            equipment.Type = "";      
+
+                            ModelState.Clear();
+                            TryValidateModel(equipment);
+                            if (ModelState.IsValid)
+                            {
+                                _context.Equipment.Add(equipment);
+                                recEquipmentCount += 1;
+                                result.Messages.Add("Equipment Added.");
+                            }
+                            else
+                            {
+                                result.Success = false;
+                                result.ErrorMessage.Add($"Invalid Equipment values Error(s): {GetModelErrors(ModelState)} ");
+                            }
+
+                            if(!string.IsNullOrWhiteSpace(r.Key1))
```

Comment:
If it isn't valid at this point, don't do any more of these adds. That should remove the multiple error messages as well. Might be able to clean these up by moving them into their own methods, then don't call that method if success is false.